app/javascript/packages/validated-field/validated-field-element.ts

```diff
@@ -138,13 +137,12 @@ class ValidatedFieldElement extends HTMLElement {
   }
 
   private getErrorElement(): HTMLElement | null {
-    const describedByElementIds = this.descriptorId?.split(' ') || [];
+    const describedByElementIds = this.input?.getAttribute('aria-describedby')?.split(' ') || [];
 
     return (
       describedByElementIds
         .map((id) => document.getElementById(id))
-        .find((element) => element?.classList.contains('usa-error-message')) ||
-      this.querySelector('.usa-error-message')
+        .find((element) => element?.classList.contains('usa-error-message')) || null
```

Comment:
I think I see the issue you're describing, at least in how it impacts something like Memorable Date, which has a specific [shared error message container](https://github.com/18F/identity-idp/blob/a10cd96f8f409d5b57249d1036c889d878439c39/app/components/memorable_date_component.html.erb#L122) that's expected to be used for error messages. It was mostly intentional that this function wouldn't find an error element when rendered in a valid state and always create a new one, with the assumption that there would be no existing error element nor `aria-describedby` present. That obviously won't work so well for how we want to use Memorable Date.

A couple ideas on how we might approach this:

1. We could use the new `error-id` attribute on `lg-validated-field` instead of the input's `aria-describedby`, so that we'd have a guaranteed reference to the error message container in Memorable Date
   - (Note: I still need to add the attribute to the Memorable Date markup)
2. Or, separately I'd been contemplating whether we could remove the error message element while the field is valid, to avoid dealing with the layout issues which prompted `display: none`. This is also made difficult by Memorable Date's requirements of an external error message, though maybe we could implement some idea of "error parent", as a configurable override to the default behavior of [appending to the input wrapper](https://github.com/18F/identity-idp/blob/d12c08afc8bbc9861de68d5b3182cbb2b3b3e15a/app/javascript/packages/validated-field/validated-field-element.ts#L133).

Reply:
> This change looks like it will cause `getOrCreateErrorMessageElement` to always create a new error element when the element was initially rendered without an error, despite the fact that the error element was already present. Both error elements will have identical IDs, but the newly created one will appear second & may not be selectable by its ID.

This should be fixed now in recent commits, via the first of the ideas listed in my previous comment.